### ISSUE TITLE
ListBox: not focusable by keyboard

### ISSUE THREAD

**Issue by idzmag** [primefaces/primereact#8200]
### Describe the bug

This is a copy of closed issue #6743 and affects accessibility.
Entry of ListBox is focusable using the Tab key only once (after the page is loaded). It is impossible to access the same ListBox a second time using the Tab or Shift+Tab.
This happens because TabIndex is set to -1 after first focus.

### Reproducer

https://stackblitz.com/edit/vitejs-vite-dtc2y2

### System Information

```Shell
System:
    OS: macOS 15.5
    CPU: (10) arm64 Apple M2 Pro
    Memory: 69.67 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.17.0 - ~/.nvm/versions/node/v20.17.0/bin/node
    Yarn: 1.22.22 - /usr/local/bin/yarn
    npm: 10.8.2 - ~/.nvm/versions/node/v20.17.0/bin/npm
  Browsers:
    Chrome: 138.0.7204.184
    Safari: 18.5
  npmPackages:
    primereact: ^10.9.6 => 10.9.6 
    react: ^18.3.1 => 18.3.1
```

### Steps to reproduce the behavior

1. Click/Focus on element before ListBox.
2. Press Tab on keyboard (OK - focus is on ListBox)
3. Press Tab on keyboard (focus is on next element)
4. Press Shift+Tab on keyboard (NOT OK - ListBox is omitted)
5. Press Tab on keyboard (NOT OK - ListBox is omitted)

### Expected behavior

You can focus the ListBox multiple times using Tab or Shift + Tab.
